src/Extensions/OrderExtension.php

```diff
@@ -248,53 +313,94 @@ public function formatDataForFlow()
         $orderItems = $this->owner->OrderItems();
 
+        $validOrderItems = false;
+
         /** @var OrderItem $orderItem */
         foreach ($orderItems as $orderItem) {
-            // Get SKU or forecast group
-            $variation = ComplexProductVariation::get()->byID($orderItem->PurchasableID);
+            $product = $orderItem->Purchasable();
+            $sku = '';
+            $quantity = $orderItem->Quantity;
+            $price = $orderItem->getBasePrice()->getDecimalValue();
+
+            // Get the SKU
+            if ($product instanceof ComplexProductVariation) {
+                $sku = $product->SKU ? $product->SKU : $product->Product()->ForecastGroup;
+
+//                echo $sku;
+                $option = $product->ProductAttributeOptions()->filter([
+                    'ProductAttribute.Title' => 'Pack'
+                ])->exclude([
+                    'Title' => '1'
+                ])->first();
+
+                // Additionally, if this is a Pack, change the quantity to match the number of bottles
+                if ($option) {
+                    // Try and get the number of items, as an int
+                    if (is_numeric($option->Title)) {
+                        $packSize = (int)$option->Title;
+                        // A pack of 6 should pass through 6 bottles
+                        // And the unit price should be for 1 bottle
+                        if ($packSize > 1) {
+                            $quantity = $orderItem->Quantity * $packSize;
+                            $price = $price / $packSize;
+                        }
+                    }
+                }
 
-            if ($variation && $variation->exists()) {
-                // TODO: Case for purchasing non-variation
-//            $sku = $purchaseable instanceof ComplexProductVariation ? $purchaseable->SKU : $purchaseable->ForecastGroup;
+            } elseif ($product->ForecastGroup) {
+                $sku = $product->ForecastGroup;
+            }
 
+            if ($sku) {
                 // Order lines
                 $orderLine = $xmlOrder->addChild('orderLines');
 
-                $orderLine->addChild('ProductCode', $variation->SKU);
-                $orderLine->addChild('Quantity', (string)$orderItem->Quantity);
                 try {
-                    $orderLine->addChild('Price', (string)$orderItem->getBasePrice()->getDecimalValue());
+                    $orderLine->addChild('ProductCode', $sku);
+                    $orderLine->addChild('Quantity', (string)$quantity);
+                    $orderLine->addChild('Price', (string)$price);
+                    $validOrderItems = true;
                 } catch (Exception $e) {
                     throw new FlowException($e->getMessage(), $e->getCode());
                 }
-            } else {
-                // Order lines
-                $orderLine = $xmlOrder->addChild('orderLines');
+            } elseif (method_exists($product, 'getFlowIdentifier')) {
```

Comment:
instead of putting this into a trait and using `method_exists` try using an interface. Then you can use `if ($product insanceof FlowProduct)` :)